pkg/s3utils/utils.go

```diff
@@ -84,6 +84,9 @@ func IsVirtualHostSupported(endpointURL url.URL, bucketName string) bool {
 // AmazonS3Host - regular expression used to determine if an arg is s3 host.
 var AmazonS3Host = regexp.MustCompile("^s3[.-]?(.*?)\\.amazonaws\\.com$")
 
+// AmazonS3ChinaHost - regular expression used to determine if the arg is s3 china host.
+var AmazonS3ChinaHost = regexp.MustCompile("^s3[.-]?(.*?)\\.amazonaws\\.com\\.cn$")
```

Comment:
I think not only the period in `[.-]?` needs to be escaped, but the `?` also needs to be removed. We must have a `.` or `-` in the match. However, `?` matches zero or one of the preceding character. So, not having either one of `[.-]`, passes the match check and even if there is an offending character in between `s3` and the region name becomes ok and captured by the `(.*?)`.
Here is an example of this kind of a failure:
Try it with `"s3xcn-north-1.amazonaws.com.cn"`. With `?`, it'll return `parts: [s3xcn-north-1.amazonaws.com.cn xcn-north-1]` because of the `?`.

Reply:
Oh yes you are right .

Reply:
@harshavardhana , AmazonS3Host regex also needs to be corrected. Would be good to add negative test cases such as the one above.